glib/src/gstring.rs

```diff
@@ -617,10 +617,10 @@ impl AsRef<[u8]> for GString {
 }
 
 impl Deref for GString {
-    type Target = str;
+    type Target = GStr;
```

Comment:
I guess we should think about the advantages / disadvantages of this. What does a `&GStr` bring in comparison to just having a `&str`?

Reply:
Do we have functions taking `GStr` only?

Reply:
Not yet, see my other comments

Reply:
Maybe better to use `str` for the time being until we make the switch then.

Reply:
I'll take this out, but https://github.com/gtk-rs/gtk3-rs/pull/739 and https://github.com/gtk-rs/gir/pull/1327 can still be merged without this if you want. They don't break anything.

Reply:
Sounds like a good idea regardless, yes. More future-proof.